### PR TITLE
Feature/wlts 24 add english phrase unique constraint

### PR DESCRIPTION
**Pull request by sem-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/WLTS-24

### Change description ###
Add unique constraint and index to `Dictionary.english_phrase` field in database.
Also make db write operations transactional

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/translate/controllers/DictionaryControllerIT.java [hunk 1]
```diff
@@ -106,11 +98,76 @@ void shouldReturn200WhenLegalRequestIsSubmitted() throws Exception {
...
+            // THEN
+            assertThatDataIntegrityViolationExceptionWasThrown(thrown);
+        }
+    }
+
+
+    @Nested
+    @Disabled // Pending WLTS-25
+    class PutDictionary {
+
```
[the remark this answers is not in the document]
Comment (sem-hmcts) [reply src/integrationTest/java/uk/gov/hmcts/reform/translate/controllers/DictionaryControllerIT.java:178]: Done in f1f1d36bade439cefa2081e191f3a6d88837c6c7

File: src/integrationTest/resources/sql/add-Dictionary.sql [hunk 2]
```diff
@@ -0,0 +1,2 @@
+insert into dictionary values
+  (1, 'English phrase', null, null);
```
Comment (mattnayler) [inline src/integrationTest/resources/sql/add-Dictionary.sql:2]: Should we bump the `dictionary_id_gen` sequence if inserting outside of call? 
```
ALTER SEQUENCE dictionary_id_seq RESTART WITH 2;
```

File: src/integrationTest/java/uk/gov/hmcts/reform/translate/controllers/DictionaryControllerIT.java [hunk 3]
```diff
@@ -121,31 +126,103 @@ void shouldReturn200WhenLegalRequestIsSubmitted() throws Exception {
...
+                                           .accept(APPLICATION_JSON_VALUE)
+                                           .content(payload));
+            }
+        };
     }
 
     @Nested
     @DisplayName("Put Dictionary")
     class PutDictionary {
+        private static final String SERVICE_AUTHORIZATION = "ServiceAuthorization";
```
Comment (mattnayler) [inline src/integrationTest/java/uk/gov/hmcts/reform/translate/controllers/DictionaryControllerIT.java:173]: Feel free to use static import of:
```
uk.gov.hmcts.reform.translate.security.SecurityUtils.SERVICE_AUTHORIZATION
```

Which in turn should probably equal a static import of: 
```
uk.gov.hmcts.reform.authorisation.filters.ServiceAuthFilter.AUTHORISATION
```

**Review by mattnayler — APPROVED** [review]
👍  Approved with two optional comments.

File: src/integrationTest/java/uk/gov/hmcts/reform/translate/controllers/DictionaryControllerIT.java (hunk 3 above)
Comment (mattnayler) [reply src/integrationTest/java/uk/gov/hmcts/reform/translate/controllers/DictionaryControllerIT.java:173]: ... ie.e. import of `uk.gov.hmcts.reform.authorisation.filters.ServiceAuthFilter.AUTHORISATION` may cause confsuion with other auth header `org.springframework.http.HttpHeader.AUTHORIZATION`

Comment (mattnayler) [reply src/integrationTest/java/uk/gov/hmcts/reform/translate/controllers/DictionaryControllerIT.java]: Adjusted in 6745da253740d36b7e1fc787428db9a78b52c4fe

**Pull request merged**